### ISSUE TITLE
When receiving emails a MUI error/warning window pops up saying "MUI_Redraw() was called from task..."

### ISSUE THREAD

**Issue by jens-maus** [jens-maus/yam#484]
**Originally on 2013-12-25 12:03:39 +0100**

___
## Summary
When using YAM 2.9 with MUI4 on AmigaOS4 and receiving emails, potentially the following error might come up:

```
MUI_Redraw() was called from task ... instead of the application's main task!

This is a violation of basic MUI rules!
```

## Steps to reproduce
 1. configure a multiple number of accounts in YAM and try to receive mails from these accounts several time.
 2. retry to do the same with enabled preselection to better trigger the problem.

## Expected results
The mails should be received without such an error/warning

## Actual results
The mails are received, but the requester pops up and suspends transfer for a second or two.

## Notes
The error seems to be depending on runtime conditions. Thus, it does not always and every time pop up. So depending how busy YAM is or how many emails are received the requester pops up or not. That might be the actual reason why it slipped into the 2.9 release.

It is not clear yet whether "preselection" is required to trigger this warning or if it also occurs if no preselection is enabled at all.

**Comment by tboeckel** [comment]
**Originally on 2013-12-29 02:30:15 +0100**

___
In (ef17493):
 * YAM_UT.c: changed the DisplayStatistics() function to be thread safe. It could happen that a POP3 mail transfer was initiated while the Incoming folder was not the currently active folder and hence its index was flushed from memory before. The POP3 download would then trigger an index reload for the Incoming folder which in turn would update the global folder statistics. That would eventually happen from a different thread than the main thread and produce a warning requester with recent MUI version. The thread safety now closes #484.

**Comment by jens-maus** [comment]
**Originally on 2014-04-06 16:32:56 +0200**

___
**Release Note:**
fixed a thread safety issue which resulted in GUI operations being executed not from the main YAM thread but from a different thread. This caused MUI to popup a warning under certain circumstances when downloading mails.
